mm2src/coins/lp_coins.rs

```diff
@@ -2433,10 +2435,52 @@ pub trait BalanceTradeFeeUpdatedHandler {
     async fn balance_updated(&self, coin: &MmCoinEnum, new_balance: &BigDecimal);
 }
 
+#[derive(Clone)]
+pub struct MmCoinStruct {
+    pub inner: MmCoinEnum,
+    is_available: Arc<AtomicBool>,
+}
+
+impl MmCoinStruct {
+    fn new(coin: MmCoinEnum) -> Self {
+        Self {
+            inner: coin,
+            is_available: AtomicBool::new(true).into(),
+        }
+    }
+
+    /// Gets the current state of the parent coin whether
+    /// it's available for the external requests or not.
+    ///
+    /// Always `true` for child tokens.
+    pub fn is_available(&self) -> bool {
+        // Tokens are always active or disabled
+        // Check if we are processing the platform coin
+        self.inner.ticker() != self.inner.platform_ticker() || self.is_available.load(AtomicOrdering::SeqCst)
+    }
+
+    /// Makes the coin disabled to the external requests.
+    /// Useful for executing `disable_coin` on parent coins
+    /// that have child tokens enabled.
+    ///
+    /// Ineffective for child tokens.
+    pub fn update_is_available(&self, to: bool) {
+        if self.inner.ticker() != self.inner.platform_ticker() {
```

Comment:
this will require additional function for every coin/token we support on mm2, which I thougt it's an overkill

Reply:
It needs a single implementation in `lp_coins`. It would also make the code more clear since what this comparison is doing is not clear at first sight, but it's not a must, you decide.